From a54dad6a16b2b09eef5e2cc6c4cb6fbd957afa20 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Sat, 10 Aug 2019 12:14:02 +0800
Subject: [PATCH 03/13] Rule audit_rules_privileged_commands_crontab add oval
 check

For rule audit_rules_privileged_commands_crontab add oval check

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../oval/wrlinux1019.xml                           | 46 ++++++++++++++++++++++
 1 file changed, 46 insertions(+)
 create mode 100644 linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_crontab/oval/wrlinux1019.xml

diff --git a/linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_crontab/oval/wrlinux1019.xml b/linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_crontab/oval/wrlinux1019.xml
new file mode 100644
index 0000000..e56b716
--- /dev/null
+++ b/linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_crontab/oval/wrlinux1019.xml
@@ -0,0 +1,46 @@
+<def-group>
+  <definition class="compliance" id="audit_rules_privileged_commands_crontab" version="1">
+    <metadata>
+      <title>Ensure auditd Collects Information on the Use of Privileged Commands - crontab</title>
+      <affected family="unix">
+        <platform>WRLinux</platform>
+	<platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>Audit rules about the information on the use of crontab is enabled.</description>
+    </metadata>
+
+    <criteria operator="OR">
+
+      <!-- Test the augenrules case -->
+      <criteria operator="AND">
+        <extend_definition comment="audit augenrules" definition_ref="audit_rules_augenrules" />
+        <criterion comment="audit augenrules crontab" test_ref="test_audit_rules_privileged_commands_crontab_augenrules" />
+      </criteria>
+
+      <!-- Test the auditctl case -->
+      <criteria operator="AND">
+        <extend_definition comment="audit auditctl" definition_ref="audit_rules_auditctl" />
+        <criterion comment="audit auditctl crontab" test_ref="test_audit_rules_privileged_commands_crontab_auditctl" />
+      </criteria>
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="audit augenrules crontab" id="test_audit_rules_privileged_commands_crontab_augenrules" version="1">
+    <ind:object object_ref="object_audit_rules_privileged_commands_crontab_augenrules" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="object_audit_rules_privileged_commands_crontab_augenrules" version="1">
+    <ind:filepath operation="pattern match">/etc/audit/rules\.d/.*\.rules</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*-a[\s]+always,exit[\s]+-F[\s]+path=\/usr\/bin\/crontab[\s]+-F[\s]+perm=x[\s]+-F[\s]+auid>=1000[\s]+-F[\s]+auid!=4294967295[\s]+(-k[\s]+|-F[\s]+key=)[\S]+[\s]*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="audit auditctl crontab" id="test_audit_rules_privileged_commands_crontab_auditctl" version="1">
+    <ind:object object_ref="object_audit_rules_privileged_commands_crontab_auditctl" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="object_audit_rules_privileged_commands_crontab_auditctl" version="1">
+    <ind:filepath>/etc/audit/audit.rules</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*-a[\s]+always,exit[\s]+-F[\s]+path=\/usr\/bin\/crontab[\s]+-F[\s]+perm=x[\s]+-F[\s]+auid>=1000[\s]+-F[\s]+auid!=4294967295[\s]+(-k[\s]+|-F[\s]+key=)[\S]+[\s]*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+</def-group>
-- 
1.9.1

